File: programmers/코딩테스트 고득점 Kit/힙(Heap)/더 맵게.cpp
#include <string>
#include <vector>
#include <queue>
#include <functional>
using namespace std;
priority_queue<int, vector<int>, greater<int>> pq;
int solution(vector<int> scoville, int K) {
	int answer = 0;
	for (int i = 0; i < scoville.size(); i++)
		pq.push(scoville[i]);
	while (!pq.empty()) {
		int one = pq.top();
		pq.pop();

		if (one >= K) break;
		else answer++;

		if (!pq.empty()) {
			int two = pq.top();
			pq.pop();
			pq.push(one + 2 * two);
		}
		else {
			answer = -1;
			break;
		}
	}
	return answer;
}

File: programmers/코딩테스트 고득점 Kit/힙(Heap)/더 맵게.java
import java.util.*;
class Solution {
    static Queue<Integer> priorityQueue = new PriorityQueue<>();
    public int solution(int[] scovilleList, int K) {
        int answer = 0;
        for(int scovile : scovilleList) {
            priorityQueue.add(scovile);
        }
        while(priorityQueue.size() >= 2 && priorityQueue.peek() < K) {
            answer++;
            priorityQueue.add(priorityQueue.remove() + priorityQueue.remove() * 2);
            if(priorityQueue.size() == 1) {
                if(priorityQueue.peek() >= K) return answer;
                else return -1;
            }
        }
        return answer;
    }
}